services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/FlowHelper.groovy

```diff
@@ -73,31 +78,31 @@ class FlowHelper {
                 false, false, "autotest flow", null, null)
     }
 
+    /**
+     * Adds flow with checking rules on source and destination switches.
+     * It is supposed if rules are installed on source and destination switches, the flow is completely created.
+     */
+    FlowPayload addFlow(FlowPayload flow, boolean checkStatus = true, boolean checkRules = true) {
+        log.debug("Adding flow '${flow.id}'")
+        def response = northbound.addFlow(flow)
+
+        !checkStatus ?: Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
+        !checkRules ?: checkRulesOnSwitches(db.getFlow(flow.id), RULES_INSTALLATION_TIME, true)
+
+        return response
+    }
+
     /**
      * Deletes flow with checking rules on source and destination switches.
      * It is supposed if rules absent on source and destination switches, the flow is completely deleted.
      */
-    FlowPayload deleteFlow(String flowId) {
+    FlowPayload deleteFlow(String flowId, boolean checkRules = true) {
         def flowEntry = db.getFlow(flowId)
 
         log.debug("Deleting flow '$flowId'")
         def response = northbound.deleteFlow(flowId)
 
-        def cookies = [flowEntry.left.cookie, flowEntry.right.cookie]
-        def switches = [flowEntry.left.sourceSwitch, flowEntry.left.destinationSwitch].toSet()
-        switches.each { sw ->
-            Wrappers.wait(RULES_DELETION_TIME) {
-                try {
-                    assert !northbound.getSwitchRules(sw).flowEntries*.cookie.containsAll(cookies)
-                } catch (HttpClientErrorException exc) {
-                    if (exc.rawStatusCode == 404) {
-                        log.warn("Switch '$sw' was not found when checking rules after flow deletion")
-                    } else {
-                        throw exc
-                    }
-                }
-            }
-        }
+        !checkRules ?: checkRulesOnSwitches(flowEntry, RULES_DELETION_TIME, false)
 
         return response
```

Comment:
I'd rather put some additional verifications here. We check the rules already, but I'd also check if northbound if not able to get that flow anymore, knowing the overall Kilda stability.
Also, it would be nice to check if flow is no longer to be found in Neo.

I would not insist on this one, but as long as it helps us to catch problems early, I'd have done it.